lib/command.coffee

```diff
@@ -23,14 +23,33 @@ module.exports = class Command
 		_.each(options.options, @option, this)
 		_.extend(this, _.omit(options, 'options'))
 
+	applyPermissions: (callback) ->
+		return callback() if not @permission?
+
+		permissionFunction = state.permissions[@permission]
+
+		if not permissionFunction?
+			error = new Error("Permission not found: #{@permission}")
+			return callback(error)
+
+		permissionFunction.call(this, callback)
+
+	# TODO: Tested implicitly by execute() tests, but it might
+	# worth to test this explicitly to find other corner cases
+	_parseOptions: (options) ->
+		allOptions = _.union(state.globalOptions, @options)
+		parsedOptions = parse.parseOptions(allOptions, options)
+
 	execute: (args = {}, callback) ->
 		params = @signature.compileParameters(args.command)
-		allOptions = _.union(state.globalOptions, @options)
-		parsedOptions = parse.parseOptions(allOptions, args.options)
-		@action.call(this, params, parsedOptions, callback)
+		parsedOptions = @_parseOptions(args.options)
+
+		@applyPermissions (error) =>
+			return callback?(error) if error?
+			@action.call(this, params, parsedOptions, callback)
```

Comment:
This can just be `@action(params, parsedOptions, callback)` since it is a function on the `this` object, so manually setting `this` to `this` is unnecessary.